lib/runtime/src/ops/element_binary.cc

```diff
@@ -1,735 +1,239 @@
+  auto input_rhs = acc.get_tensor<Permissions::RO>(RHS_INPUT);
+
+  auto output_grad = acc.get_tensor_grad<Permissions::RO>(OUTPUT);
+  auto input_lhs_grad = acc.get_tensor_grad<Permissions::RW>(LHS_INPUT);
+  auto input_rhs_grad = acc.get_tensor_grad<Permissions::RW>(RHS_INPUT);
+
+  return profile(backward_kernel,
+                 profiling,
+                 "[ElementBinary] backward_time = %.2lfms\n",
+                 per_device_state,
+                 output_grad.get_float_ptr(),
+                 input_lhs.get_float_ptr(),
+                 input_rhs.get_float_ptr(),
+                 input_lhs_grad.get_float_ptr(),
+                 input_rhs_grad.get_float_ptr(),
+                 attrs.type,
+                 attrs.should_broadcast_lhs,
+                 attrs.should_broadcast_rhs);
+}
+
+static void backward_task(Task const *task,
+                          std::vector<PhysicalRegion> const &regions,
+                          Context ctx,
+                          Runtime *runtime) {
+  TaskArgumentAccessor acc(task, regions, ctx, runtime);
+  backward_task_impl(acc);
+}
+
+CostMetrics
+    measure_operator_cost(SimEnvFactory const &sim,
+                          ElementBinaryAttrs const &attrs,
+                          InputParallelTensorDesc const &input_shape_lhs,
+                          InputParallelTensorDesc const &input_shape_rhs,
+                          ProfilingSettings const &settings,
+                          MachineView const &mv) {
+  auto env = sim.new_environment();
+
+  ParallelTensorShape output_shape =
+      get_output_shape(attrs, input_shape_lhs.shape, input_shape_rhs.shape);
+
+  SimTaskBinding init_binding;
+  init_binding.bind(LHS_INPUT, input_shape_lhs);
+  init_binding.bind(RHS_INPUT, input_shape_rhs);
+  init_binding.bind(OUTPUT, output_shape);
+  init_binding.bind_arg(ATTRS, attrs);
+  init_binding.bind_arg(HANDLE, ff_handle());
+
+  auto init_accessor =
+      env.get_init_accessor(ELEMENTBINARY_INIT_TASK_ID, init_binding);
+  DeviceSpecific<ElementBinaryPerDeviceState> per_device_state =
+      init_task_impl(init_accessor);
+
+  SimTaskBinding fwd_binding;
+  fwd_binding.bind(LHS_INPUT, input_shape_lhs);
+  fwd_binding.bind(RHS_INPUT, input_shape_rhs);
+  fwd_binding.bind(OUTPUT, output_shape);
+
+  fwd_binding.bind_arg(PROFILING, settings);
+  fwd_binding.bind_arg(PER_DEVICE_STATE, per_device_state);
+
+  SimTaskBinding bwd_binding = infer_bwd_binding(fwd_binding);
+
+  auto fwd_accessor =
+      env.get_fwd_accessor(ELEMENTBINARY_FWD_TASK_ID, fwd_binding);
+  auto bwd_accessor =
+      env.get_bwd_accessor(ELEMENTBINARY_BWD_TASK_ID, bwd_binding);
+
+  float forward_time = forward_task_impl(fwd_accessor).value();
+  float backward_time = backward_task_impl(bwd_accessor).value();
+
+  float sync_time = default_estimate_sync_time(env);
+  return make_metrics(forward_time, backward_time, sync_time, env);
+}
+
+template <>
+void register_task<ELEMENTBINARY_INIT_TASK_ID>() {
```

Comment:
Check out [this code in my PR](https://github.com/flexflow/FlexFlow/pull/1097/files#diff-8c4e9b314279391e71c120c443da8ee470be5ed8feee33ae3b09914c032bb070) on splitting the signature from registering task (useful for testing when we don't need to register the task, but we want to get the signature).